opentelemetry/src/main/scala/zio/telemetry/opentelemetry/baggage/Baggage.scala

```diff
@@ -168,14 +170,14 @@ object Baggage {
         private def getCurrentContext(implicit trace: Trace): UIO[Context] =
           ctxStorage.get
 
-        private def modifyBuilder(body: BaggageBuilder => BaggageBuilder)(implicit trace: Trace): UIO[Context] =
+        private def modifyBuilder(body: BaggageBuilder => BaggageBuilder)(implicit trace: Trace): URIO[Scope, Context] =
           modifyContext { ctx =>
             body(Baggaje.fromContext(ctx).toBuilder)
               .build()
               .storeInContext(ctx)
           }
 
-        private def modifyContext(body: Context => Context)(implicit trace: Trace): UIO[Context] =
+        private def modifyContext(body: Context => Context)(implicit trace: Trace): URIO[Scope, Context] =
```

Comment:
(Question): I'm wondering what if we change it to:
```
private def modifyContext(body: Context => Context)(implicit trace: Trace): URIO[Context] =
  ZIO.scoped(ctxStorage.updateAndGet(body))
```
Because anyway, we call `context.makeCurrent` inside `.updateAndGet` each time.
This way we can avoid changing Baggage API so it stays backward-compatible.